### ISSUE TITLE
egl config creation failure with eglInitialize shim

### ISSUE THREAD

**Issue by FakeShell** [libhybris/libhybris#567]
it appears with latest hybris, some apps in debian 13 are failing to open and/or have no gpu acceleration
the commit that broke it is https://github.com/libhybris/libhybris/commit/f42bb5605ee23b8a2baa76b85e5bc706dc575d74

an example of the error message:
```
[GFX1-]: Fallback WR to SW-WR, forced (t=2.1757) |[6][GFX1-]: Failed to create EGLConfig for WebRender! (t=1.09959) |[7][GFX1-]: Failed to create EGLConfig for WebRender! (t=1.10359) |[8][GFX1-]: Failed GL context creation for hardware WebRender: true (t=1.10359) |[9][GFX1-]: Failed to create EGLConfig for WebRender! (t=2.1557) |[10][GFX1-]: Failed to create EGLConfig for WebRender! (t=2.1597) |[11][GFX1-]: Failed GL context creation for hardware WebRender: true (t=2.1597) |[12][GFX1-]: Failed to create EGLConfig for WebRender! (t=2.1637) |[13][GFX1-]: Failed to create EGLConfig for WebRender! (t=2.1677) |[14][GFX1-]: Failed GL context creation for hardware WebRender: true (t=2.1677) |[15][GFX1-]: Failed to create EGLConfig for WebRender! (t=2.1717) [GFX1-]: Fallback WR to SW-WR, forced
```

tested with firefox 128

**Comment by mlehtima** [comment]
Just to be sure that you tested the very latest version which has changes fixing issues introduced in the commit you linked?

**Comment by FakeShell** [comment]
Yes this is including the very last commit (50838b52123d0d3271b2895c2649fef9f713d5ac f4ad41e89ae9d1ebb93f27769de4ea2d92516fd3 cbca34e66e39d13aea414e113848ebb865feb3ad)

**Comment by krnlyng** [comment]
Hm i don't see what could cause this. Except there might be one thing (but i'm not 100% certain if it can cause this):

Here: https://github.com/libhybris/libhybris/blob/master/hybris/egl/egl.c#L255 we are calling android's eglGetDisplay with EGL_DEFAULT_DISPLAY regardless of which display is being passed in to the function via display_id. Later we are returning real_display to the native side.
What this means is that hybris_egl_display_get_mapping(), regardless of inputs, will always return the real_display that corresponds to eglGetDisplay(EGL_DEFAULT_DISPLAY) (since eglDefaultDisplay always returns the same display with the same argument). This effectively means that the first call to eglGetDisplay(dpy_1) will force the display for all later calls to be the display dpy_1.
Unfortunately in android there is only EGL_DEFAULT_DISPLAY: https://github.com/DolbyLaboratories/aosp-platform-frameworks-native/blob/master/opengl/libs/EGL/egldefs.h#L30 so we can't specify a different one in https://github.com/libhybris/libhybris/blob/master/hybris/egl/egl.c#L255

One solution to this might be to instead of creating a mapping for the displays is to instead return the actual WaylandDisplay to the native side instead of real_display and then to hook every method that uses EGLDisplay and before passing that EGLDisplay to the corresponding android side convert it to the android display. So no HYBRIS_IMPLEMENT_FUNCTION stuff.

Unfortunately i can't really test this as i don't have a setup for this. And it might not be the real solution, but it seems it's the most likely to me.

**Comment by FakeShell** [comment]
this ended up happening because of mali MEOW. after getting rid of meow entirely and just using libGLES_mali it works fine now. not sure why it causes issues but apparently this is not the first time meow does something funky (as evident in https://github.com/libhybris/libhybris/pull/543)

i will close this issue since we don't have any issues with this anymore but feel free to reopen if anyone is interested in actually investigating.
